<launch>

    <node pkg="hbba_lite" type="on_off_hbba_filter_node" name="beat_detector_node_filter">
        <remap from="in" to="audio_input_signed_16_16000"/>
        <remap from="out" to="beat_detector/audio_input_signed_16_16000_filtered"/>
        <remap from="filter_state" to="beat_detector/filter_state"/>
    </node>
    <node pkg="audio_utils" type="beat_detector_node" name="beat_detector_node">
        <param name="sampling_frequency" value="16000"/>
        <param name="frame_sample_count" value="64"/>
        <param name="oss_fft_window_size" value="1024"/>
        <param name="flux_hamming_size" value="15"/>
        <param name="oss_bpm_window_size" value="1024"/>
        <param name="min_bpm" value="50"/>
        <param name="max_bpm" value="180"/>
        <param name="bpm_candidate_count" value="4"/>

        <remap from="audio_in" to="beat_detector/audio_input_signed_16_16000_filtered"/>
        <remap from="bpm" to="dance/bpm"/>
        <remap from="beat" to="dance/beat"/>
    </node>

</launch>
